python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition.py

```diff
@@ -407,26 +408,47 @@ def evaluate(self, context: "AutomationContext") -> "AutomationResult":
         )
 
 
-class AutomationResult(NamedTuple):
-    condition: AutomationCondition
-    condition_unique_id: str
-    value_hash: str
+class AutomationResult:
+    """The result of evaluating an AutomationCondition."""
+
+    context: "AutomationContext"
+    true_slice: AssetSlice
+    child_results: Sequence["AutomationResult"]
 
     start_timestamp: float
     end_timestamp: float
```

Comment:
even thought it is word, would prefer to encapsulate these so that they are read-only so people can't mutate the result object

Reply:
updated